frontend/src/components/organisms/ViewEvents.tsx

```diff
@@ -83,55 +86,42 @@ const UpcomingEvents = () => {
   if (isLoading) return <Loading />;
 
   /** Error screen */
-  if (isError) return <Error />;
+  if (isError) return <FetchDataError />;
+
+  console.log(upcomingEventsSupervisor);
 
   return (
     <div>
-      {role === "Supervisor" ? (
-        <div>
-          <h2>
-            Yay!! you're a supervisor. You get special privileges on this page.
-          </h2>
-        </div>
-      ) : role === "Admin" ? (
-        <div>
-          <h2>
-            Yay!! you're an admin. You get special privileges on this page.
-          </h2>
-        </div>
-      ) : role === "Volunteer" ? (
-        <div>
-          <h2>You're a volunteer</h2>
-        </div>
-      ) : (
-        <></>
+      {(role === "Supervisor" || role === "Admin") && (
+        <Link href="/events/create">
+          <Button className="mb-2 w-full sm:w-max">Create New Event</Button>
+        </Link>
       )}
-      <Link href="/events/create">
-        <Button className="mb-2 w-full sm:w-max">Create New Event</Button>
-      </Link>
       {/* Display when no events are found */}
       {/* TODO: make this look better */}
-      {upcomingEventsSupervisor.length == 0 && upcomingEventsVolunteer == 0 && (
+      {((role === "Admin" && upcomingEventsSupervisor.length == 0) ||
+        (role === "Supervisor" && upcomingEventsSupervisor.length == 0) ||
+        (role === "Volunteer" && upcomingEventsVolunteer == 0)) && (
         <div className="p-10">
           <div className="text-center">You have no upcoming events</div>
         </div>
       )}
       {/* List of Upcoming events user supervises */}
-      {upcomingEventsSupervisor.map((event: ViewEventsEvent) => (
-        <div>
-          <div className="mt-5" />
-          <EventCardNew key={event.id} event={event} />
-        </div>
-      ))}
-
+      {(role === "Supervisor" || role === "Admin") &&
+        upcomingEventsSupervisor.map((event: ViewEventsEvent) => (
+          <div>
+            <div className="mt-5" />
+            <EventCardNew key={event.id} event={event} />
+          </div>
+        ))}
       {/* List of Upcoming events user registered for */}
-      {upcomingEventsVolunteer.map((event: ViewEventsEvent) => (
-        <div>
-          <div className="mt-5" />
-          <EventCardNew key={event.id} event={event} />
-        </div>
-      ))}
-
+      {role === "Volunteer" &&
+        upcomingEventsVolunteer.map((event: ViewEventsEvent) => (
+          <div>
+            <div className="mt-5" />
+            <EventCardNew key={event.id} event={event} />
+          </div>
+        ))}
```

Comment:
Follows from my earlier question on whether this is necessary, if you register for an event, you should be able to View the event details. This applies to admin and supervisors that might want to register for events

Reply:
I added the conditional here because I think from partner calls they mentioned that supervisors don't register for events. Also the event details are still visible, this conditional is strictly for the registration card in the bottom right.

Reply:
Ah right, I wasn't able to see this because of the error I kept seeing on the view events page.